servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancerBuilder.java

```diff
@@ -177,7 +177,23 @@ private static final class DefaultLoadBalancerFactory<ResolvedAddress, C extends
         public <T extends C> LoadBalancer<T> newLoadBalancer(String targetResource,
              Publisher<? extends Collection<? extends ServiceDiscovererEvent<ResolvedAddress>>> eventPublisher,
              ConnectionFactory<ResolvedAddress, T> connectionFactory) {
-            return new DefaultLoadBalancer<ResolvedAddress, T>(id, targetResource, eventPublisher,
+            throw new IllegalStateException("Generic constructor not supported by " +
```

Comment:
Consider using `UnsupportedOperationException`

Reply:
Good call, that's a much better exception.